6-AdvancedScenarios/3-call-api-acrs/API/routes/adminRoutes.js

```diff
@@ -1,58 +1,47 @@
 const express = require('express');
 const dashboardController = require('../controllers/adminController');
+const router = express.Router();
 
+const authProvider = require("../auth/AuthProvider");
+const { protectedResources } = require("../authConfig");
 
-module.exports = (authProvider) => {
+function isAuthenticated(req, res, next) {
+    if (!req.session.isAuthenticated) {
+        return res.redirect('/admin/signin'); // redirect to sign-in route
+    }
 
-    const router = express.Router();
+    next();
+}
 
-    // main routes
-    router.get('/', (req, res, next) => res.redirect('/admin/home'));
-    router.get('/home', dashboardController.getHomePage);
+// main routes
+router.get('/', (req, res, next) => res.redirect('/admin/home'));
+router.get('/home', dashboardController.getHomePage);
 
-    // authentication routes
-    router.get('/signin', authProvider.signIn({ postLoginRedirect: '/admin' }));
-    router.get('/signout', authProvider.signOut({ postLogoutRedirect: '/admin' }));
+// authentication routes
+router.get('/signin', (req, res, next) =>  authProvider.login(req, res, next));
```

Comment:
Can you update the `login` method's signature so that we can use it here like

`router.get('/signin', authProvider.login())`

(same for logout, getToken etc)

Reply:
Done!